internal/command/views/plan.go

```diff
@@ -398,18 +412,62 @@ func renderChangesDetectedByRefresh(before, after *states.State, schemas *terraf
 	// If we only have changes regarding deposed instances, or the diff
 	// renderer is suppressing irrelevant changes from the legacy SDK, there
 	// may not have been anything to display to the user.
-	if len(diffs) > 0 {
+	if len(diffs) > 0 || len(summaries) > 0 {
 		view.streams.Print(
 			view.colorize.Color("[reset]\n[bold][cyan]Note:[reset][bold] Objects have changed outside of Terraform[reset]\n\n"),
 		)
-		view.streams.Print(format.WordWrap(
-			"Terraform detected the following changes made outside of Terraform since the last \"terraform apply\":\n\n",
-			view.outputColumns(),
-		))
+		if len(diffs) > 0 {
+			if len(summaries) > 0 {
+				view.streams.Print(format.WordWrap(
+					"Terraform detected the following changes made outside of Terraform since the last \"terraform apply\", which may have caused proposed changes below:\n\n",
+					view.outputColumns(),
+				))
+			} else {
+				view.streams.Print(format.WordWrap(
+					"Terraform detected the following changes made outside of Terraform since the last \"terraform apply\":\n\n",
+					view.outputColumns(),
+				))
+			}
+			for _, diff := range diffs {
+				view.streams.Print(diff)
+			}
+			switch mode {
+			case plans.RefreshOnlyMode:
+				view.streams.Println(format.WordWrap(
+					"\nThis is a refresh-only plan, so Terraform will not take any actions to undo these. If you were expecting these changes then you can apply this plan to record the updated values in the Terraform state without changing any remote objects.",
+					view.outputColumns(),
+				))
+			default:
+				view.streams.Println(format.WordWrap(
+					"\nUnless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or respond to these changes.",
+					view.outputColumns(),
+				))
+			}
 
-		for _, diff := range diffs {
-			view.streams.Print(diff)
 		}
+		if len(summaries) > 0 {
+			if len(diffs) > 0 {
+				view.streams.Print(format.WordWrap(
+					"\nTerraform also changes to some other resource instances which don't seem to affect the plan:\n",
```

Comment:
Hi @apparentlymart, not sure if this made it into our main repo PR (#30486). I checked and couldn't see it, but could be wrong. If it didn't, then ignore me :)

Missing a "detected" in the string. Should be "...Terraform also detected changes..."